registry/spec.md

```diff
@@ -1491,71 +1632,1350 @@ An attempt to delete all versions MUST generate an error.
+      "plural": "schemagroups",
+      "schema": "TBD",
+      "resources": [
+        {
+          "singular": "schema",
+          "plural": "schemas",
+          "versions": -1,
+        }
+      ]
+    }
+  ]
+}
+```
+
+#### Group: schemagroups
+
+The group (GROUP) name for the Schema Registry is `schemagroups`. The group does
+not have any specific extension attributes.
+
+A schema group is a collection of schemas that are related to each other in some
+application-defined way. A schema group does not impose an restrictions on the
+contained schenmas, meaning that a schema group can contain schemas of different
+formats. Every schema MUST reside inside a schema group.
+
+Example:
+
+``` meta
+{
+  "schemagroups": {
+    "com.example.schemas": {
+      "type": "schemagroup",
+      "id": "com.example.schemas",
+      "schemas": {
+        ...
+      }
+    }
+  }
+}
+```
+
+#### Resource: schemas
+
+The resources (RESOURCE) collection inside of schema groups is named `schemas`.
+The type of the resource is `schema`. Any single `schema` is a container for
+one or more `versions`, which hold the concrete schema documents or schema
+document references.
+
+Any new schema version that is added to a schema definition MUST be backwards
+compatible with all previous versions of the schema, meaning that a consumer
+using the new schema MUST be able to understand data encoded using a prior
+version of the schema. If a new version introduces a breaking change, it MUST be
+regisetered as a new schema with a new name.
+
+When you retrieve a schema without qualifying the version, you will get the
+latest version of the schema, see [retrieving a
+resource](#retrieving-a-resource). The latest version is the lexically greatest
+version number, whereby all version ids MUST be left-padded with spaces to the
+same length before being compared.
+
+The following extension is defined for the `schema` object in addition to the
+basic [attributes](#attributes-and-extensions):
+
+##### `format` (Schema format)
+
+- Type: String  
+- Description: Identifies the schema format. In absence of formal media-type
+  definitions for several important schema formats, we define a convention here
+  to reference schema formats by name and version as `{NAME}/{VERSION}`. This
+  specification defines a set of common [schema format names](#schema-formats)
+  that MUST be used for the given formats, but applications MAY define
+  extensions for other formats on their own.
+- Constraints:
+  - REQUIRED
+  - MUST be a non-empty string
+  - MUST follow the naming convention `{NAME}/{VERSION}`, whereby `{NAME}` is
+    the name of the schema format and `{VERSION}` is the version of the schema
+    format in the format defined by the schema format itself.
+- Examples:
+  - `JsonSchema/draft-07`
+  - `Protobuf/3`
+  - `XSD/1.1`
+  - `Avro/1.9`  
+
+#### Resource Version: schemaversion
+
+The `VERSION` object of the `schema` resource is of type `schemaversion`. The
```

Comment:
(if we keep `type`)
Looking at the text here and the example below, I wonder if this will confuse people because the `schema` type is actually a group, not a singular thing - but its name isn't `schemagroup`. That's already taken. My first reaction upon reading this section was that the "type" really should be `schema` for both but, of course, if we keep `type` we can't do that. Something feels off here and I want to blame the presence of `type` :-)

Reply:
we need to discuss the relationship between ID, Name and Version. The sample uses the ID as the version string which doesn't seem consistent.